crates/chain/src/indexed_tx_graph.rs

```diff
@@ -0,0 +1,406 @@
+use core::{convert::Infallible, ops::AddAssign};
+
+use bitcoin::{OutPoint, Script, Transaction, TxOut};
+
+use crate::{
+    keychain::Balance,
+    sparse_chain::ChainPosition,
+    tx_graph::{Additions, TxGraph, TxNode},
+    BlockAnchor, ChainOracle, FullTxOut, ObservedIn, TxIndex,
+};
+
+/// An outwards-facing view of a transaction that is part of the *best chain*'s history.
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub struct TxInChain<'a, T, A> {
+    /// Where the transaction is observed (in a block or in mempool).
+    pub observed_in: ObservedIn<&'a A>,
+    /// The transaction with anchors and last seen timestamp.
+    pub tx: TxNode<'a, T, A>,
+}
+
+/// A structure that represents changes to an [`IndexedTxGraph`].
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(
+    feature = "serde",
+    derive(serde::Deserialize, serde::Serialize),
+    serde(
+        crate = "serde_crate",
+        bound(
+            deserialize = "A: Ord + serde::Deserialize<'de>, IA: serde::Deserialize<'de>",
+            serialize = "A: Ord + serde::Serialize, IA: serde::Serialize"
+        )
+    )
+)]
+#[must_use]
+pub struct IndexedAdditions<A, IA> {
+    /// [`TxGraph`] additions.
+    pub graph_additions: Additions<A>,
+    /// [`TxIndex`] additions.
+    pub index_additions: IA,
```

Comment:
I did not understand what this field implies. And why it needs to be a generic. And seems like its an "spk index addition", something that implies change in the spk indexing system? If that's the case a better doc would help for it.

Reply:
Yes it is a way for the index to persist something after it has done a scan by the user re-loading `IndexedTxAdditions` into their `IndexedTxGraph`. I think we can't avoid this because we made it so `KeychainTxOutIndex` can increment its revealed index after a scan via look ahead mechanism.